sdk/python/test/e2e/utils.py

```diff
@@ -17,13 +18,19 @@ def verify_unschedulable_job_e2e(
     logging.info(f"\n\n\n{job_kind} is creating")
     client.wait_for_job_conditions(name, namespace, job_kind, {constants.JOB_CONDITION_CREATED})
 
-    # Job should have a Created condition.
-    if not client.is_job_created(name, namespace, job_kind):
-        raise Exception(f"{job_kind} should be in Created condition")
+    logging.info("Checking 3 times that pods have been pending for a while...")
+    for num in range(3):
+        logging.info(f"Number of attempts: {int(num)+1}/3")
+        # Job should have a Created condition.
+        if not client.is_job_created(name, namespace, job_kind):
+            raise Exception(f"{job_kind} should be in Created condition")
 
-    # Job shouldn't have a Running condition.
-    if client.is_job_running(name, namespace, job_kind):
-        raise Exception(f"{job_kind} shouldn't be in Running condition")
+        # Job shouldn't have a Running condition.
+        if client.is_job_running(name, namespace, job_kind):
```

Comment:
Can you explain what you mean by "pending for a while"? Are you referring to a situation which is in created but not running?  If then, job will get into running state after retry?

Reply:
I meant unschedulable pods (gang scheduling).

Reply:
If before the training-operator updates the job condition from `Runnng=false` to `Running=true`. this test code gets the job condition and the job condition has `Running=false` or doesn't have `Running` condition, this test unintended passes. 

So, let's imagine the following situation:

Current e2e:

1. Test: Deploy job with gang scheduling setting (.runPolicy.schedulingpolicy).
2. Operator: Failed to set schedulerName=volcano or unintended incorrect PodGroup.
3. Test: Get the job with `Running=false` or without `Running` condition
4. Pods: Pods are immediately scheduled to Node and start since the job doesn't have appropriate gang scheduling settings.
5. Operator: Update the job condition with `Running=true`. 
6. Test: Success! (Unintended)

Reply:
Note: This `verify_unschedulable_job_e2e` function verifies that gang scheduler integration works well.

Reply:
Thanks for the explanation.